src/main/java/br/com/dandrade/viagens/controllers/dto/input/CompanyRequest.java

```diff
@@ -0,0 +1,36 @@
+package br.com.dandrade.viagens.controllers.dto.input;
+
+import br.com.dandrade.viagens.models.Company;
+import br.com.dandrade.viagens.models.Country;
+
+import javax.validation.constraints.NotBlank;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class CompanyRequest {
+
+    @NotBlank
+    private String name;
+    @NotBlank
+    private String country;
```

Comment:
Aqui eu realmente não especifiquei bem e você fez o que achou melhor, acho que está tudo bem. Mas pode considerar que o front vai passar o id do país que você vai relacionar. Pensa que lá no front a aplicação vai passar o id do país. Dessa forma você nem vai precisar validar se nome do país existe.

Reply:
Verdade! Fiquei com duvida nessa parte.
Vou corrigir.

Reply:
Na minha opinião só precisa de um not null no id. Quero dizer, se vier um id que não existe vai dar um pau de optional vazia em algum lugar... De uma forma ou de outra(validando ou aceitando a possibilidade da exception de empty explodir), você ta protegido.

Reply:
entendi.